### DISCUSSION TITLE
Failed to get storage metrics

### DISCUSSION THREAD

**Discussion by ninuxio** [k3s-io/k3s#11303] [General]
Hi,

My logs are full of

`477 metrics.go:299] "Failed to get storage metrics" storage_cluster_id="etcd-0" err="context deadline exceeded"`

I run a K3s cluster version 1.31.2 with kine.

it seems to be generated by the Apiserver ?

Happy to provide more infos if needed

**Comment by JSN-1** [comment]
i have the same issue, it started after upgrading from k3s 1.29.x to 1.30.x, problem still persist after upgrading to 1.31.4

metrics-server is running fine and kubectl top pods | nodes is working as well.

message is written in journal log 5 - 6 times per minute causing higher load on k3s-server.

```
Jan 06 19:50:24 <redacted> k3s[335232]: I0106 19:50:24.818644  335232 metrics.go:299] "Failed to get storage metrics" storage_cluster_id="etcd-0" err="context deadline exceeded"
Jan 06 19:50:26 <redacted> k3s[335232]: I0106 19:50:26.654560  335232 metrics.go:299] "Failed to get storage metrics" storage_cluster_id="etcd-0" err="context deadline exceeded"
Jan 06 19:50:31 <redacted> k3s[335232]: I0106 19:50:31.717152  335232 metrics.go:299] "Failed to get storage metrics" storage_cluster_id="etcd-0" err="context deadline exceeded"
Jan 06 19:50:36 <redacted> k3s[335232]: I0106 19:50:36.068181  335232 metrics.go:299] "Failed to get storage metrics" storage_cluster_id="etcd-0" err="context deadline exceeded"
```

**Comment by JSN-1** [comment]
created a new thread https://github.com/k3s-io/k3s/discussions/11526 as this one is closed.
